### ISSUE TITLE
Either I once again did something wrong, or no support for Resource Loader? [User Error]

### ISSUE THREAD

**Issue by Flabort** [crazysnailboy/VillagerTrades#6]
[Resource Loader](https://minecraft.curseforge.com/projects/resource-loader)

So as per resource loader instructions (I don't want to go through the work of making a resource pack), I made a folder in the resource folder called "hobo", I put a en_us.lang file in there and a folder called "assets", and I put in the two files I wanted for textures for the custom villager in there. 

in case it's my code that I messed up, here it is, but if it's that the resource loader cannot provide resources for my custom villager, then I'd like to make an [enhancement] feature request.

```
{
  profession: {
    name: "vtt:banker"
    texture: "hobo:assets/banker.png",
    zombieTexture: "hobo:assets/zombie_banker.png",
  }
  careers: [
    "banker"
  ]
}
```

```
{
  Profession:"vtt:banker",
  career:"banker",

  Offers:{
    Recipes:[
      {action:"add",CareerLevel:1,buy:{id:"ctdcurrency:hundredth",Count:{min:5,max:7}},sell:{id:"ctdcurrency:twentheth",Count:1}},
      {action:"add",CareerLevel:1,buy:{id:"ctdcurrency:twentheth",Count:1},sell:{id:"ctdcurrency:hundredth",Count:{min:3,max:5}}},
      {action:"add",CareerLevel:1,buy:{id:"ctdcurrency:twentheth",Count:{min:2,max:3}},sell:{id:"ctdcurrency:tenth",Count:1}},
      {action:"add",CareerLevel:1,buy:{id:"ctdcurrency:tenth",Count:1},sell:{id:"ctdcurrency:twentheth",Count:{min:1,max:2}}},
      {action:"add",CareerLevel:2,buy:{id:"ctdcurrency:tenth",Count:{min:1,max:2}},buy2:{id:"ctdcurrency:twentheth",Count:1}, sell:{id:"ctdcurrency:quarter",Count:1}},
      {action:"add",CareerLevel:2,buy:{id:"ctdcurrency:twentheth",Count:{min:5,max:7}},sell:{id:"ctdcurrency:quarter",Count:1}},
      {action:"add",CareerLevel:2,buy:{id:"ctdcurrency:quarter",Count:1},sell:{id:"ctdcurrency:twentheth",Count:{min:3,max:5}}},
      {action:"add",CareerLevel:2,buy:{id:"ctdcurrency:quarter",Count:{min:4,max:5}},sell:{id:"ctdcurrency:one",Count:1}},
      {action:"add",CareerLevel:2,buy:{id:"ctdcurrency:one",Count:1},sell:{id:"ctdcurrency:twentheth",Count:{min:3,max:4}}},
      {action:"add",CareerLevel:3,buy:{id:"ctdcurrency:one",Count:{min:5,max:7}},sell:{id:"ctdcurrency:five",Count:1}},
      {action:"add",CareerLevel:3,buy:{id:"ctdcurrency:five",Count:1},sell:{id:"ctdcurrency:one",Count:{min:3,max:5}},
      {action:"add",CareerLevel:3,buy:{id:"ctdcurrency:five",Count:{min:2,max:3}},sell:{id:"ctdcurrency:ten",Count:1}},
      {action:"add",CareerLevel:3,buy:{id:"ctdcurrency:ten",Count:1},sell:{id:"ctdcurrency:five",Count:{min:1,max:2}},
      {action:"add",CareerLevel:4,buy:{id:"ctdcurrency:ten",Count:{min:2,max:3}},sell:{id:"ctdcurrency:twenty",Count:1}},
      {action:"add",CareerLevel:4,buy:{id:"ctdcurrency:twenty",Count:1},sell:{id:"ctdcurrency:ten",Count:{min:1,max:2}},
      {action:"add",CareerLevel:4,buy:{id:"ctdcurrency:twenty",Count:{min:5,max:7}},buy2{id:"ctdcurrency:ten",Count:1},sell:{id:"ctdcurrency:fifty",Count:1}},
      {action:"add",CareerLevel:4,buy:{id:"ctdcurrency:ten",Count:{min:5,max:7}},sell:{id:"ctdcurrency:fifty",Count:1}},
      {action:"add",CareerLevel:4,buy:{id:"ctdcurrency:fifty",Count:1},sell:{id:"ctdcurrency:ten",Count:{min:3,max:5}}
    ]
  }
}
```

**Comment by Flabort** [comment]
Nevermind, I see what I did wrong. Missplaced a comma when defining the textures.

**Comment by Flabort** [comment]
Nope. Fixing those commas gave me a purple/black villager with no trades. And the lang file didn't take. :(

**Comment by Flabort** [comment]
Well, I've got it to this, but that part still isn't working:
```
{
  Profession:"vtt:banker",
  Career:"banker",

  Offers:{
    Recipes:[
      {action:"add",CareerLevel:1,buy:{id:"ctdcurrency:hundredth",Count:{min:5,max:7}},sell:{id:"ctdcurrency:twentheth",Count:1}},
      {action:"add",CareerLevel:1,buy:{id:"ctdcurrency:twentheth",Count:1},sell:{id:"ctdcurrency:hundredth",Count:{min:3,max:5}}},
      {action:"add",CareerLevel:1,buy:{id:"ctdcurrency:twentheth",Count:{min:2,max:3}},sell:{id:"ctdcurrency:tenth",Count:1}},
      {action:"add",CareerLevel:1,buy:{id:"ctdcurrency:tenth",Count:1},sell:{id:"ctdcurrency:twentheth",Count:{min:1,max:2}}},
      {action:"add",CareerLevel:2,buy:{id:"ctdcurrency:tenth",Count:{min:1,max:2}},buyB:{id:"ctdcurrency:twentheth",Count:1}, sell:{id:"ctdcurrency:quarter",Count:1}},
      {action:"add",CareerLevel:2,buy:{id:"ctdcurrency:twentheth",Count:{min:5,max:7}},sell:{id:"ctdcurrency:quarter",Count:1}},
      {action:"add",CareerLevel:2,buy:{id:"ctdcurrency:quarter",Count:1},sell:{id:"ctdcurrency:twentheth",Count:{min:3,max:5}}},
      {action:"add",CareerLevel:2,buy:{id:"ctdcurrency:quarter",Count:{min:4,max:5}},sell:{id:"ctdcurrency:one",Count:1}},
      {action:"add",CareerLevel:2,buy:{id:"ctdcurrency:one",Count:1},sell:{id:"ctdcurrency:twentheth",Count:{min:3,max:4}}},
      {action:"add",CareerLevel:3,buy:{id:"ctdcurrency:one",Count:{min:5,max:7}},sell:{id:"ctdcurrency:five",Count:1}},
      {action:"add",CareerLevel:3,buy:{id:"ctdcurrency:five",Count:1},sell:{id:"ctdcurrency:one",Count:{min:3,max:5}}},
      {action:"add",CareerLevel:3,buy:{id:"ctdcurrency:five",Count:{min:2,max:3}},sell:{id:"ctdcurrency:ten",Count:1}},
      {action:"add",CareerLevel:3,buy:{id:"ctdcurrency:ten",Count:1},sell:{id:"ctdcurrency:five",Count:{min:1,max:2}}},
      {action:"add",CareerLevel:4,buy:{id:"ctdcurrency:ten",Count:{min:2,max:3}},sell:{id:"ctdcurrency:twenty",Count:1}},
      {action:"add",CareerLevel:4,buy:{id:"ctdcurrency:twenty",Count:1},sell:{id:"ctdcurrency:ten",Count:{min:1,max:2}}},
      {action:"add",CareerLevel:4,buy:{id:"ctdcurrency:twenty",Count:{min:5,max:7}},buyB:{id:"ctdcurrency:ten",Count:1},sell:{id:"ctdcurrency:fifty",Count:1}},
      {action:"add",CareerLevel:4,buy:{id:"ctdcurrency:ten",Count:{min:5,max:7}},sell:{id:"ctdcurrency:fifty",Count:1}},
      {action:"add",CareerLevel:4,buy:{id:"ctdcurrency:fifty",Count:1},sell:{id:"ctdcurrency:ten",Count:{min:3,max:5}}}
    ]
  }
}
```

**Comment by crazysnailboy** [comment]
Hi,

Yeah I'm sorry about the Buy and BuyB stuff and the brittle casing - the syntax for the mod is based on that used by the summon command and so the casing has to be consistent with what Minecraft expects the NBT data for a villager to be, and NBT tag names are case sensitive...

Profession and Career in the villager file and the trade file should be lower-case, so if you try
```
  profession:"vtt:banker",
  career:"banker",
``` 
that should work.

As for resource loader I've not tested it, but if you've got the texture working then the lang file should probably work too. Do you know what the pack format used by resource loader is, and can you set it? if it's 2 then the file should be called "en_US.lang", and if it's 3 then it should be "en_us.lang". This mod uses 2 so it's internal lang file is called "en_US.lang".

**Comment by Flabort** [comment]
Thanks. I finally got it working, every part of it. Mostly it was trial and error and complaining out loud/to you, but with it working now, I am very happy. 

https://imgur.com/a/mCnqD

First two images were from before I got the lang file working, last two are after I got lang and trades working.

**Comment by crazysnailboy** [comment]
Nice :)

What's that HUD mod you're using in the first image? It doesn't seem to be picking up the villager name from the lang file.

Also, could I improve the wiki documentation to make this easier?

**Comment by Flabort** [comment]
WAILA, and I took that screenshot before I got the lang file working. It works properly now.
I used lowercase for en_us.lang, and what I had done wrong was put it in the wrong folder - I didn't realize there needed to be a lang folder.

So the new file path is hobocraft/resources/vtt/lang/en_us.lang for the language file. And likewise hobocraft/resources/vtt/textures/entity/villager/banker.png, which goes in the config like vtt:textures/entity/vilager/banker.png

Sure, a small improvement to the wiki documentation could be had for those unfamiliar with resource packs. Specifically the things I learned are "Textures need to be in a folder named 'textures'", "lang files need to be in a folder named 'lang'", and "the mod Resource Loader works in place of a resource pack, but textures and lang files still need to be in a folder named 'vtt'."
